catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/formulaeditor/FormulaEditorAddVariableTest.java

```diff
@@ -88,4 +88,19 @@ public void addListTest() {
 		onDataList()
 				.onListAtPosition(0).checkHasName(listName);
 	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void cancelOnTouchCancelButtonTest() {
+		final String varName1 = "canceled";
+		final String varName2 = "variable";
+		onDataList()
+				.performCancel(varName1);
+
+		onDataList()
+				.performAdd(varName2);
+
+		onDataList().onVariableAtPosition(0)
+				.checkHasName(varName2);
```

Comment:
I feel like it would make much more sense to only call "performCancel" and then check if the variable does not exist.
You do not check if the cancellation worked right now, you only check if adding worked. If pressing cancel would add the variable anyway this test still wouldn't fail.